receiver/mongodbreceiver/config.go

```diff
@@ -0,0 +1,61 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbreceiver
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	configtls.TLSClientSetting              `mapstructure:"tls,omitempty"`
+	Hosts                                   []confignet.NetAddr `mapstructure:"hosts"`
+	Username                                string              `mapstructure:"username"`
+	Password                                string              `mapstructure:"password"`
+	ReplicaSet                              string              `mapstructure:"replica_set,omitempty"`
+	Timeout                                 time.Duration       `mapstructure:"timeout"`
```

Comment:
This is not described in the docs. 

Am I correct in thinking it will used by the mongo client library?

Reply:
Initial plan was to use it for 

https://github.com/mongodb/mongo-go-driver/blob/384560d7efa96ed08030bf1eaebba0230776b82f/mongo/options/clientoptions.go#L694-L700

and the corresponding socket option in the client.